File: java/Stack.java
public class Stack {
    private java.util.List items;
    public Stack() {
        this.items = new java.util.ArrayList();
    }
    public void push(Object item) {
        items.add(item);
    }
    public Object pop() {
        if (!isEmpty()) {
            return items.remove(items.size() - 1);
        }
        throw new IllegalStateException("pop from an empty stack");
    }
    public Object peek() {
        if (!isEmpty()) {
            return items.get(items.size() - 1);
        }
        throw new IllegalStateException("peek from an empty stack");
    }
    public boolean isEmpty() {
        return items.size() == 0;
    }
    public int size() {
        return items.size();
    }
}

File: python/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        if not self.isEmpty():
            return self.items.pop()
        raise IndexError("pop from an empty stack")

    def peek(self):
        if not self.isEmpty():
            return self.items[-1]
        raise IndexError("peek from an empty stack")

    def isEmpty(self):
        return len(self.items) == 0

    def size(self):
        return len(self.items)